Tests.Integration.Presentation.Web/References/ExternalReferencesTest.cs

```diff
@@ -0,0 +1,98 @@
+
+            //Assert
+            AssertCreatedReference(_title, createdReference, _externalReferenceId, _referenceUrl, _display);
+            systemDto = await ItSystemHelper.GetSystemAsync(systemDto.Id);
+            Assert.Equal(createdReference.Id, systemDto.ReferenceId.GetValueOrDefault(-1)); //First reference must be marked as "the reference"
+        }
+
+        [Fact]
+        public async Task Can_Create_Reference_In_ItSystemUsage()
+        {
+            //Arrange
+            var systemDto = await ItSystemHelper.CreateItSystemInOrganizationAsync(A<string>(), TestEnvironment.DefaultOrganizationId, AccessModifier.Public);
+            var usageDTO = await ItSystemHelper.TakeIntoUseAsync(systemDto.Id, TestEnvironment.DefaultOrganizationId);
+
+            //Act
+            var createdReference = await ReferencesHelper.CreateReferenceAsync(_title, _externalReferenceId, _referenceUrl, _display, dto => dto.ItSystemUsage_Id = usageDTO.Id);
+
+            //Assert
+            AssertCreatedReference(_title, createdReference, _externalReferenceId, _referenceUrl, _display);
+            usageDTO = await ItSystemHelper.GetItSystemUsage(usageDTO.Id);
+            Assert.Equal(createdReference.Id, usageDTO.ReferenceId.GetValueOrDefault(-1)); //First reference must be marked as "the reference"
+        }
+
+        [Fact]
+        public async Task Can_Create_Reference_In_ItContract()
+        {
+            //Arrange
+            var contract = await ItContractHelper.CreateContract(A<string>(), TestEnvironment.DefaultOrganizationId);
+
+            //Act
+            var createdReference = await ReferencesHelper.CreateReferenceAsync(_title, _externalReferenceId, _referenceUrl, _display, dto => dto.ItContract_Id = contract.Id);
+
+            //Assert
+            AssertCreatedReference(_title, createdReference, _externalReferenceId, _referenceUrl, _display);
+            contract = await ItContractHelper.GetItContract(contract.Id);
+            Assert.Equal(createdReference.Id, contract.ReferenceId.GetValueOrDefault(-1)); //First reference must be marked as "the reference"
+        }
+
+        [Fact]
+        public async Task Can_Create_Reference_In_ItProject()
+        {
+            //Arrange
+            var project = await ItProjectHelper.CreateProject(A<string>(), TestEnvironment.DefaultOrganizationId);
+
+            //Act
+            var createdReference = await ReferencesHelper.CreateReferenceAsync(_title, _externalReferenceId, _referenceUrl, _display, dto => dto.ItProject_Id = project.Id);
+
+            //Assert
+            AssertCreatedReference(_title, createdReference, _externalReferenceId, _referenceUrl, _display);
+            project = await ItProjectHelper.GetProjectAsync(project.Id);
+            Assert.Equal(createdReference.Id, project.ReferenceId.GetValueOrDefault(-1)); //First reference must be marked as "the reference"
+        }
+
```

Comment:
Hvad med en test der laver to referencer på samme entity og tjekker det er den første der er markeret som vis?

Reply:
agreed - fixer